src/OpenDiffix.Core/Analyzer.fs

```diff
@@ -430,7 +430,25 @@ let private isMoneyStyle arg =
   | Constant (Integer c) -> Regex.IsMatch($"%i{c}", "^[125]0*$")
   | _ -> false
 
+let private isColumnReference arg =
+  match arg with
+  | ColumnReference _ -> true
+  | _ -> false
+
+let private isConstant arg =
+  match arg with
+  | Constant _ -> true
+  | _ -> false
+
 let private validateGeneralization accessLevel expression =
+  if accessLevel <> Direct then
```

Comment:
Just FYI: after the initial merge, I had a plan to refactor these bits and move the validation to `QueryValidator`, I'll move this new bit over too (I don't remember exactly why was this troublesome at first).